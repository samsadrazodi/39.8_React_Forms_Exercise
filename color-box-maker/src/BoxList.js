import React, {useState} from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import './BoxList.css'
const BoxList =() =>{
    const [boxes, setBoxes] = useState([]);


    const addBox = (width,height, color)=>{
        setBoxes([...boxes, {width, height, color}]);
    };

    const removeBox= (index)=>{
        const newBoxes = boxes.filter((_, i) => i !== index);
        setBoxes(newBoxes);
    };

    return (
        <div>
            <div className="form">
            <NewBoxForm addBox={addBox} />
            </div>
        <div className="boxes">
        {boxes.map((box, index) => (
          <Box
            key={index}
            width={box.width}
            height={box.height}
            color={box.color}
            removeBox={() => removeBox(index)}
          />
        ))}
        </div>
      </div>
    )
}

export default BoxList;