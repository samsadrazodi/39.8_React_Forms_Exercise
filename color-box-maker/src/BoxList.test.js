import React from "react";
import {render, fireEvent} from "@testing-library/react";
import BoxList from "./BoxList";

it('renders without crashing', function (){
    render(<BoxList />);
});

it('matches snapshot', function (){
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});


it("should add a blue div with specified width and height", () => {
    const { getByLabelText, getByText, container } = render(<BoxList />);
  
    const widthInput = getByLabelText("Width:");
    const heightInput = getByLabelText("Height:");
    const colorInput = getByLabelText("Color:");
    const submitButton = getByText("Add Box");
  
    fireEvent.change(widthInput, { target: { value: "200" } });
    fireEvent.change(heightInput, { target: { value: "200" } });
    fireEvent.change(colorInput, { target: { value: "blue" } });
    fireEvent.click(submitButton);
  
    // Find the added blue box within the rendered container
    const blueBox = container.querySelector(".box");
  
    expect(blueBox).toBeInTheDocument(); // Check if the blue box is in the document
    expect(blueBox).toHaveStyle("width: 200px");
    expect(blueBox).toHaveStyle("height: 200px");
    expect(blueBox).toHaveStyle("background-color: blue");
  });
  
  
  
  
  