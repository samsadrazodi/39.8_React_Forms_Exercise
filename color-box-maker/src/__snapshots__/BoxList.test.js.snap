// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form"
    >
      <form>
        <div>
          <label
            for="width"
          >
            Width:
          </label>
          <input
            id="width"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            for="height"
          >
            Height:
          </label>
          <input
            id="height"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            for="color"
          >
            Color:
          </label>
          <input
            id="color"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Add Box
        </button>
      </form>
    </div>
    <div
      class="boxes"
    />
  </div>
</DocumentFragment>
`;
